### ISSUE TITLE
Does not work with Require.js

### ISSUE THREAD

**Issue by incompl** [videojs/generator-videojs-plugin#193]
Require.js thinks that the `video.js` external in `umd.rollup.config.js` is a script itself and is trying to load it from current path. I'm not sure how to fix or work around this issue.

**Comment by BrandonOCasey** [comment]
I have a working example with contrib-ads here: https://jsfiddle.net/uy9680os/57/

**Comment by gkatsev** [comment]
I think the change here with `external` usage on rollup which puts in declared dependencies in AMD is a breaking change for users. Also, the requirement on using `map` for rjs is an unnecessary burden.

**Comment by BrandonOCasey** [comment]
Yeah I think we need to keep this issue open, but for now there is a work-around.

**Comment by gkatsev** [comment]
So, looking into why this is happening and I think it's because rollup's `globals` is broken and potentially doesn't match our expectations.
The UMD bundle it generates will always require/define the dependencies whether `globals` is set or not but our expectations based on browserify and browserify-shim is that once something is set to `global`, it will *always* load from that shimmed location (i.e., global/window object).

A workaround/quick fix would be to use something like https://github.com/patrickhulce/rollup-plugin-shim or https://github.com/rollup/rollup-plugin-alias to shim `video.js` and other "globals" into small files/strings that just return `window.videojs` etc.

It would also be nice if we can figure out why globals isn't working in rollup and whether our expectations match rollup's or not.

**Comment by BrandonOCasey** [comment]
https://github.com/videojs/generator-videojs-plugin/pull/200
